### ISSUE TITLE
[Bug]: dependency version mismatch (utf-8-validate) (non-blocking)

### ISSUE THREAD

**Issue by grmkris** [MetaMask/metamask-sdk#821]
### SDK

Web

### Provide environment information

when installing dependency using pnpm

### MetaMask SDK Version

0.14.3

### MetaMask Mobile app Version

0

### Describe the Bug

```
┬ @metamask/sdk 0.14.3
      ├─┬ @metamask/sdk-communication-layer 0.14.3
      │ └─┬ socket.io-client 4.7.5
      │   └─┬ engine.io-client 6.5.3
      │     └─┬ ws 8.11.0
      │       └── ✕ unmet peer utf-8-validate@^5.0.2: found 6.0.3 in @metamask/sdk-communication-layer
      └─┬ @metamask/sdk-install-modal-web 0.14.1
        └─┬ react-i18next 13.5.0
          └── ✕ unmet peer i18next@">= 23.2.3": found 22.5.1 in @metamask/sdk-install-modal-web
```

seeing this when installing using pnpm, any chance you update this? it is not blocking, but would love to avoid these warnings in my project

### Expected Behavior

there should be no warnings

### To Reproduce

install latest wagmi, that used metamask-sdk under the hood using pnpm

**Comment by KeeganEad** [comment]
Possibly related? I'm getting stuck in an install loop with npm.  Logs if i run `npm install --verbose --timing` otherwise it just hangs at a certain point.  

I was able to successfully install with `--legacy-peer-deps`

```
npm timing idealTree:node_modules/@wagmi/connectors/node_modules/@metamask/sdk Completed in 4ms
npm timing idealTree:node_modules/@metamask/sdk-install-modal-web Completed in 1ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/qr-code-styling Completed in 1ms
npm timing idealTree:node_modules/qrcode-generator Completed in 0ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/@wagmi/connectors/node_modules/@metamask/sdk Completed in 5ms
npm timing idealTree:node_modules/@metamask/sdk-install-modal-web Completed in 1ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/qr-code-styling Completed in 0ms
npm timing idealTree:node_modules/qrcode-generator Completed in 0ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/@wagmi/connectors/node_modules/@metamask/sdk Completed in 5ms
npm timing idealTree:node_modules/@metamask/sdk-install-modal-web Completed in 1ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/qr-code-styling Completed in 0ms
npm timing idealTree:node_modules/qrcode-generator Completed in 0ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/@wagmi/connectors/node_modules/@metamask/sdk Completed in 4ms
npm timing idealTree:node_modules/@metamask/sdk-install-modal-web Completed in 1ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/qr-code-styling Completed in 1ms
npm timing idealTree:node_modules/qrcode-generator Completed in 0ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/@wagmi/connectors/node_modules/@metamask/sdk Completed in 5ms
npm timing idealTree:node_modules/@metamask/sdk-install-modal-web Completed in 1ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/qr-code-styling Completed in 1ms
npm timing idealTree:node_modules/qrcode-generator Completed in 0ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/@wagmi/connectors/node_modules/@metamask/sdk Completed in 4ms
npm timing idealTree:node_modules/@metamask/sdk-install-modal-web Completed in 1ms
npm timing idealTree:node_modules/i18next Completed in 0ms
npm timing idealTree:node_modules/qr-code-styling Completed in 0ms
npm timing idealTree:node_modules/qrcode-generator Completed i
```

**Comment by alantaylordesign** [comment]
I'm having the same npm install loop issue as described by @KeeganEad.  Noticed it first yesterday afternoon.

**Comment by aLANparty** [comment]
I'm having the same npm install loop issue as described by @KeeganEad. Noticed it first yesterday afternoon.

**Comment by christopherferreira9** [comment]
Hi @grmkris ! Can you please update the MetaMask SDK to the latest version `0.18.6` and see if the problem persists?

**Comment by aLANparty** [comment]
Updating to `0.18.6` does not fix the issue, but I spent a little time and was able to track down where the problem is.  The real issue is in `@metamask/sdk-install-modal-web` and has to do with some conflicting dependencies in the `react-i18next` package.  Currently `@metamask/sdk-install-modal-web` has a peer dependency on both `react-i18next@^13.2.2` and `i18next@22.5.1`, but `react-i18next@13.2.2` itself has a peer dependency on `i18next >= 23.2.3`.  `@metamask/sdk` also has a dependency on `i18next@22.5.1`, but this wouldn't be a problem if `@metamask/sdk-install-modal-web` didn't require `react-i18next@^13.2.2`.

So to fix the issue either `@metamask/sdk-install-modal-web` needs it's `react-i18next` dependency lowered to something that is compatible with `i18next@22.5.1` or both `@metamask/sdk` and `@metamask/sdk-install-modal-web` need their `i18next` dep updated to be `i18next >= 23.2.3`.

To work around this issue for now I've added this override to my package.json to force the use of `i18next >= 23.2.3`.  This fixes the installation issue as it just forces the installation of i18next@^23.2.3, but I've not tested anything with it.  Don't know if this has breaking changes with your packages requesting `v22.5.1`

```
"overrides": {
    "i18next": "^23.2.3"
},
```

Here are the relevant lines from the packages I described that show where the conflict is.

## @metamask/sdk
```
"dependencies": {
    "i18next": "22.5.1",
},
```

## @metamask/sdk-install-modal-web
```
"peerDependencies": {
    "i18next": "22.5.1",
    "react-i18next": "^13.2.2",
},
```

## react-i18next
```
"peerDependencies": {
    "i18next": ">= 23.2.3",
},
```

**Comment by christopherferreira9** [comment]
Thank you for investigating @aLANparty ! We'll get back to this very shortly!

**Comment by aLANparty** [comment]
I think I should add that the original bug report described an issue with both `utf-8-validate` and `i18next`.  The issue with `utf-8-validate` *was* fixed.  Only the issue with `i18next` remains and it's what I'm describing.

**Comment by MarkAPhillips** [comment]
I am having the same issue when using the latest version of Next.js - but it seems to relate the react version peer dependency which is specific to 18.2.0 where as the latest version uses 18.3.1

`"peerDependencies": {
    "@react-native-async-storage/async-storage": "^1.19.6",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-native": "*"
  },`

**Comment by ottodevs** [comment]
> I am having the same issue when using the latest version of Next.js - but it seems to relate the react version peer dependency which is specific to 18.2.0 where as the latest version uses 18.3.1
> 
> ```
> "peerDependencies": {
>     "@react-native-async-storage/async-storage": "^1.19.6",
>     "react": "^18.2.0",
>     "react-dom": "^18.2.0",
>     "react-native": "*"
>   },
> ```
> 
> using version 0.20.2 of the React SDK

The package `@metamask/sdk-install-modal-web` seems to be the origin of some other deprecated deps like a bunch of babel ones. The main issue here is this package depends on `react-native@0.74.2`, that depends on a lot of deprecated stuff.

**Comment by omridan159** [comment]
Hey @grmkris,
This issue has been resolved in [PR #921](https://github.com/MetaMask/metamask-sdk/pull/921).
These changes will be available in the next SDK release.

Thanks!
